sourcemash/frontend/static/js/views/register_modal_view.js

```diff
@@ -0,0 +1,41 @@
+Sourcemash.Views.RegisterModalView = Backbone.View.extend({
+    template: JST['register-modal'],
+
+    events: {
+        'submit #register': 'registerFromModal',
+    },
+
+    registerFromModal: function(e) {
+        e.preventDefault();
+        var formData = JSON.stringify($("#register").serializeObject());
+        $('#register-password').val('');
+        $('#register-password-confirm').val('');
+        var posting = $.ajax({
+                          type: "POST",
+                          url: "/register",
+                          data: formData,
+                          success: this.registerOrShowErrors,
```

Comment:
Doesn't look like this can be done unfortunately. While the packet returned has error codes in the 400's, Flask-Security still sends back a 200 as long as the processing was successful. So we get a 200 whether the packet contains errors or an authenticated user, so we have to leave it as a check-for-both function.